cfwlogd/src/main.rs

```diff
@@ -242,7 +242,7 @@ fn validate_log_files(vmobjs: &Vmobjs) {
                 Ok(info) => {
                     // If we find the "\n" we add 1 to the index so that when we truncate the file
                     // we keep the newline.
-                    let idx = info.index.and_then(|i| Some(i + 1)).unwrap_or(0);
+                    let idx = info.index.map(|i| i + 1).unwrap_or(0);
```

Comment:
This looks correct to me.  Map will map the value if the Option is Some otherwise it will leave it as None.

Reply:
Could also be cleaned up to use `info.index.map_or(0, |i| i +1)`